app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -336,6 +339,25 @@ export class CloneRepository extends React.Component<
     return exists
   }
 
+  private async isPathEmptyFolder(path: string) {
+    try {
+      const directoryFiles = await readdirSync(path)
+
+      // If the path changed while we were checking, we don't care anymore.
+      if (this.state.path !== path) {
+        return
+      }
+
+      return directoryFiles.length === 0
+    } catch (error) {
+      if (error.code === -2) {
+        // Directory does not exist error
+        return false
+      }
+      return error
```

Comment:
@shiftkey Since it has been decided to not have both `isPathEmptyFolder` and `doesPathExist` and instead cover both with one function, the goal of this function may need to shift a bit.

Since this function now needs to also cover situations where the folder does not exist as well, we have a few options. If there if the path doesn't exist `readdir` will return an error. We can potentially return `true` in this case, indicating that the path is an empty folder, which is not _exactly_ correct but does not mess with the logic flow.

Otherwise we can change the question we're trying to answer a bit.

```typescript
  private async doesPathContainFiles(path: string): Promise<boolean | Error> {
    try {
      const directoryFiles = await readdir(path)
      return directoryFiles.length !== 0
    } catch (error) {
      if (error.code === 'ENOENT') {
        return false
      }

      return error
    }
  }
```
> I'd also like to avoid returning an error here - we should return something typed or rethrow the error (preferably an Error type) if it should be handled by the caller.

I'm not sure what the full variety of errors that can be thrown by `readdir` are. I would love if I could cover the `ENOENT` path does not exist error and return a boolean in that case and not have to handle an error from there, but I have done `boolean | Error` for now.

I have not committed these changes yet, as I figured it was best to discuss the changes first and see which direction it would be best to go in.